code/datums/traits/_quirk.dm

```diff
@@ -29,7 +29,16 @@
 	add()
 	if(spawn_effects)
 		on_spawn()
-		addtimer(CALLBACK(src, .proc/post_add), 30)
+	if(quirk_holder.client)
+		post_add()
+	else
+		RegisterSignal(quirk_holder, COMSIG_MOB_LOGIN, .proc/on_quirk_holder_first_login)
+
+
+///Description of what this proc does here.
```

Comment:
Do give the proc a proper documentation.
And ask if you don't understand how any of this works.
The idea is that you learn here how to make the code respond to events so it only runs a single time, not for me to just code it for you.

Reply:
The idea of unregistering it after trigger is great i did not consider it.
I was approaching this as always later to add client i never thought they could be attached at this point.

I am happy for the feedback